test/validation/api/scheduler/scheduler.c

```diff
@@ -1982,6 +1982,87 @@ static void parallel_execute(odp_schedule_sync_t sync, int num_queues,
 	reset_queues(args);
 }
 
+/* Worker creates a private schedule group and a queue in it, enqueues events,
+ * and verifies scheduled events come only from that private group. */
+static int private_group_thread(void *arg)
+{
+	thread_args_t *thr_args = arg;
+	test_globals_t *globals = thr_args->globals;
+	odp_pool_t pool = odp_pool_lookup(MSG_POOL_NAME);
+	odp_thrmask_t mask;
+	odp_schedule_group_t private_group;
+	odp_queue_param_t queue_param;
+	odp_queue_t enqueue_to, sched_from;
+	odp_event_t ev;
+	odp_buffer_t buffer;
+	uint32_t i;
+	const uint32_t num_ev = 16;
+	int thr_id;
+	char group_name[ODP_SCHED_GROUP_NAME_LEN];
+	char queue_name[ODP_QUEUE_NAME_LEN];
+
+	CU_ASSERT_FATAL(pool != ODP_POOL_INVALID);
+
+	thr_id = odp_thread_id();
+	snprintf(group_name, sizeof(group_name), "pgt_group_%d", thr_id);
+	snprintf(queue_name, sizeof(queue_name), "pgt_queue_%d", thr_id);
+
+	odp_thrmask_zero(&mask);
+	odp_thrmask_set(&mask, thr_id);
+
+	/* Create private group for this worker (only this thread in the mask) */
+	private_group = odp_schedule_group_create(group_name, &mask);
+	CU_ASSERT_FATAL(private_group != ODP_SCHED_GROUP_INVALID);
+
+	odp_queue_param_init(&queue_param);
+	queue_param.type        = ODP_QUEUE_TYPE_SCHED;
+	queue_param.sched.sync  = ODP_SCHED_SYNC_PARALLEL;
+	queue_param.sched.group = private_group;
+	queue_param.size        = num_ev;
+
+	enqueue_to = odp_queue_create(queue_name, &queue_param);
+	CU_ASSERT_FATAL(enqueue_to != ODP_QUEUE_INVALID);
+
+	/* Pre-drain */
+	odp_schedule_pause();
+	while ((ev = odp_schedule(NULL, ODP_SCHED_NO_WAIT)) != ODP_EVENT_INVALID)
+		odp_event_free(ev);
+	odp_schedule_resume();
+
+	/* Enqueue events into private queue */
+	for (i = 0; i < num_ev; i++) {
+		buffer = odp_buffer_alloc(pool);
+		CU_ASSERT_FATAL(buffer != ODP_BUFFER_INVALID);
+		CU_ASSERT_FATAL(odp_queue_enq(enqueue_to, odp_buffer_to_event(buffer)) == 0);
+	}
+
+	/* Wait for all threads to finish enqueuing before scheduling */
+	odp_barrier_wait(&globals->barrier);
+
+	/* Schedule events and verify they come only from private queue/group */
+	i = 0;
+	while (i < num_ev) {
+		ev = odp_schedule(&sched_from, odp_schedule_wait_time(ODP_TIME_MSEC_IN_NS));
+		if (ev == ODP_EVENT_INVALID)
+			continue;
+
+		CU_ASSERT(sched_from == enqueue_to);
+		CU_ASSERT(odp_queue_sched_group(sched_from) == private_group);
+		odp_event_free(ev);
+		i++;
+	}
+
+	odp_schedule_pause();
+	while ((ev = odp_schedule(NULL, ODP_SCHED_NO_WAIT)) != ODP_EVENT_INVALID)
+		odp_event_free(ev);
+	odp_schedule_resume();
```

Comment:
Could use `drain_queues()` function and check that zero is returned.

Reply:
Nothing should be returned here anymore and `drain_queues()` does the same thing.